crates/sui-framework/packages/move-stdlib/sources/u128.move

```diff
@@ -27,6 +27,17 @@ public fun diff(x: u128, y: u128): u128 {
     std::macros::num_diff!(x, y)
 }
 
+/// Calculate x * y / z, upscaling the intermediate values to avoid overflow.
```

Comment:
I feel like "upcasting" might be a more common phrasing in Move? Thoughts @damirka?